### PR TITLE
Add GraphQL dataloaders #577

### LINKED ISSUE #577: Add GraphQL dataloaders
Currently we are not leveraging GraphQL dataloaders for batching queries to database.

Add dataloaders in every @ResolveField method.

### PR DESCRIPTION
**Pull request by IAlexHilimovich**
Add GraphQL dataloaders for batching field resolvers.
Closes #577

### CODE REVIEW

File: packages/apiserver/src/data-loader-type.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const DataLoaderType = {
```
Comment (Chichmarin17) [inline packages/apiserver/src/data-loader-type.ts:1]: can be named as `DataLoaders` (like previously as we don't use it as type, could be misleading)

File: packages/apiserver/src/utils/helpersFunctions/groupByKey.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export const groupByKey = <T>(items: T[], key: string): Record<number, T[]> => {
```
Comment (Chichmarin17) [inline packages/apiserver/src/utils/helpersFunctions/groupByKey.ts:1]: we can move out this generic type and use it in other modules without need to duplicate `Record<number, Person[]>` etc

**Review by Chichmarin17 — APPROVED** [review]

File: packages/apiserver/src/data-loader-type.ts (hunk 1 above)
Comment (IAlexHilimovich) [reply packages/apiserver/src/data-loader-type.ts]: Renamed to DataLoaders

File: packages/apiserver/src/utils/helpersFunctions/groupByKey.ts (hunk 2 above)
Comment (IAlexHilimovich) [reply packages/apiserver/src/utils/helpersFunctions/groupByKey.ts]: Added GroupRecord<T> type

**Pull request merged**